### DISCUSSION TITLE
gramine app deployment

### DISCUSSION THREAD

**Discussion by reclock** [gramineproject/gramine#1672] [Q&A]
Now there is a problem. I use Gramine to compile programs on the development machine, generate signatures, and deploy them to the testing machine

1. Can Gramine start the program normally, or does it need to be recompiled on the testing machine
2. Will the measurement value change if two machines are using the same docker image
3. How to solve development and deployment issues

**Comment by dimakuv** [comment]
> Can Gramine start the program normally, or does it need to be recompiled on the testing machine

No. (Given that the Gramine version and the program version/dependencies are exactly the same. Docker images help with this.)

> Will the measurement value change if two machines are using the same docker image

No.

> How to solve development and deployment issues

Which issues exactly?
